ydb/library/yql/udfs/common/datetime2/datetime_udf.cpp

```diff
@@ -683,6 +712,23 @@ TValue DoAddYears(const TValue& date, i64 years, const NUdf::IDateBuilder& build
         }
     }
 
+    template <>
+    TUnboxedValue TSplit<TDate32>::Run(
+        const IValueBuilder* valueBuilder,
+        const TUnboxedValuePod* args) const
+    {
+        try {
+            EMPTY_RESULT_ON_EMPTY_ARG(0);
+
+            TUnboxedValuePod result(0);
+            auto& storage = Reference64(result);
+            storage.FromDate32(valueBuilder->GetDateBuilder(), args[0].Get<i32>());
+            return result;
+        } catch (const std::exception& e) {
+            UdfTerminate((TStringBuilder() << Pos_ << " " << e.what()).data());
```

Comment:
I will better add values to exceptions where they are thrown in datetime64.h

Reply:
I doubt, there are way more informative, than the current ones: at least nobody knows the timezones IDs, you returned to the user. What I was talking about earlier is [kinda this technique](https://github.com/ydb-platform/ydb/pull/8084/files#diff-4c008caaee33f8a256fdff5c4fb3681882b7b937906c499911014e069aa739dcR928-R942): use `storage.ToString` and yield the human-readable value, that allow to easily interpret the error message and fix the problem. Consider [the following notice](https://ydb.tech/docs/en/yql/reference/udf/list/datetime#converting-time-to-date) in the docs:
> It's worth mentioning that several TzDatetime or TzTimestamp values with a positive timezone offset cannot be cast to TzDate. Consider the example below:
> ```
> SELECT CAST(TzDatetime("1970-01-01T23:59:59,Europe/Moscow") as TzDate);
> /* Fatal: Timestamp 1970-01-01T23:59:59.000000,Europe/Moscow cannot be casted to TzDate */
> ```
> Starting from the Unix epoch, there is no valid value representing midnight on 01/01/1970 for the Europe/Moscow timezone. As a result, such a cast is impossible and fails at runtime.

Anyway, I see no reason to block this PR with this nit, so let's proceed with your change.

Reply:
As I understand, exceptions thrown from datetime.h and datetime64.h are some kind of internal ones, and exist mainly for debug reason in case there is some bug in make/split implementation
Values fed to DateTime::Make and DateTIme::Split functions should be already valid